packages/app/obojobo-document-engine/src/scripts/oboeditor/components/editor-app.js

```diff
@@ -123,6 +126,30 @@ class EditorApp extends React.Component {
 		this.reloadDraft(this.state.draftId, mode)
 	}
 
+	displayLockedModal() {
+		this.setState({
+			requestStatus: 'invalid',
+			requestError: {
+				title: 'Module is Being Edited.',
+				message: 'Someone else is currently editing this module, please try again later.'
+			}
+		})
+	}
+
+	startRenewEditLockInterval() {
+		setInterval(() => {
+			this.createEditLock(this.state.draftId).catch(() => {
+				this.displayLockedModal()
+			})
+		}, RENEW_LOCK_INTERVAL)
```

Comment:
Does this mean a user could leave a tab open and have a lock on a module indefinitely? Do we need to implement an inactive timeout to stop someone from doing this?

Reply:
Yea.  even backgrounded tabs will keep firing.  We could udse a cheap solution for now - perhaps using the editor events?